include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,618 @@
+        }
+        else
+        {
+            if constexpr (std::ranges::BidirectionalRange<source_range_type>)
+            { // Simply take one before the end. We can do this as we require source_range_type to be a common range.
+                back_iterator = std::ranges::prev(std::ranges::end(src_range));
+            }
+            else
+            { // For all other cases we need to set the back_iterator in linear time to the correct position.
+                back_iterator = std::ranges::begin(src_range);
+                if constexpr (std::ranges::SizedRange<source_range_type>)
+                {
+                    std::ranges::advance(back_iterator, std::ranges::size(src_range) - 1);
+                }
+                else // We don't have the size, so we need to increment until one before the end in a linear pass.
+                {
+                    auto tmp_it = back_iterator;
+                    do
+                    {
+                        back_iterator = tmp_it;
+                    } while (++tmp_it != std::ranges::end(src_range));
+                }
+            }
+        }
+    }
+
+    /*!\brief Constructs from a view.
+     * \tparam    other_range_t  The type of the range to be wrapped with seqan3::detail::pairwise_combine_view;
+     *                           must model std::ranges::ViewableRange and source_range_type must be constructible
+     *                           with other_range wrapped in view::all.
+     * \param[in] range          The source range to be wrapped.
+     *
+     * \details
+     *
+     * During construction the iterator pointing to the last element of the view is cached (not the end of the range).
+     * This optimises the call to end if the source range models only ForwardRange. Otherwise the call to end will be
+     * linear in the number of elements of the source range.
+     *
+     * \attention This view cannot be chained immediately after an infinite range, because upon construction it will
+     *            take forever to reach the last element of the view.
+     *
+     * ### Complexity
+     *
+     * Constant if `other_range_t` models std::ranges::BidirectionalRange, otherwise linear.
+     */
+    template <typename other_range_t>
+    //!\cond
+        requires !std::Same<remove_cvref_t<other_range_t>, pairwise_combine_view> &&
+                 std::ranges::ViewableRange<other_range_t> &&  // Must come after self type check to avoid conflicts with the move constructor.
+                 std::Constructible<source_range_type, ranges::ref_view<std::remove_reference_t<other_range_t>>>
+            //TODO: Investigate: the following expression is equivalent to the one above but raises a weird assertion in
+            //      the ranges adaptor suggesting that the pairwise_combine_view is not a ViewableRange.
+            //      std::Constructible<source_range_type, decltype(view::all(std::declval<other_range_t &&>()))>
```

Comment:
Don't know if that was the issue, but did you try to `remove_reference_t<other_range_t>`? Otherwise reference collapsing will remove your `&&` here.

Reply:
The `&&` does exactly what it should. It gives me a glvalue that I pass to view::all. 
Btw. the decltype is identical with the one above. But I believe there is a weird cycle between the constructor definition and the constraint using the view::all adaptor, which might invoke the constructor again or something. I wasted a lot of time on this and the current implementation is used by the C++20 views per draft as well. Note, that neither the ranges views nor our old views implement this and there will be enough time to rereview this later. But for now, I'd really like to get this ... PR done 🙏